src/js/trove/iolib.js

```diff
@@ -0,0 +1,55 @@
+({
+  requires: [ ],
+  provides: {
+    shorthands: { },
+    values: {
+      // Intentional design decision to not expose as a pure function. This is different from Haskell.
+      //
+      // ```haskell
+      // getChar :: IO Char
+      // ```
+      //
+      // ref: https://www.haskell.org/tutorial/io.html
+      "prompt": ["arrow", ["String"], "String"]
+    },
+    aliases: { },
+    datatypes: { }
+  },
+  nativeRequires: ["readline"],
+  theModule: function(RUNTIME, NAMESPACE, uri, readline) {
+    function Input() {
+        return RUNTIME.pauseStack(function(restarter) {
+          const rl = readline.createInterface({ input: RUNTIME.stdin });
+          
+          // input does not need to display anything
+          new Promise(resolve => rl.question('', input => resolve(input)))
+            .then(result => restarter.resume(RUNTIME.makeString(result)))
+            .catch(error => {
+              // TODO: we should probably NOT fail this hard
+              restarter.error(RUNTIME.makeString(error));
```

Comment:
Not 100% sure how to trigger this so it fails. Started looking at mocking, unsure how that will play with the harness as is, may need to design it to be a bit more robust. Is a failure test something desired in a v1 implementation?